internal/hooktest/hook.go

```diff
@@ -22,14 +22,14 @@ func (*testHook) Version() int {
 
 // RunHook implementation
 func (*testHook) PreRun(_ context.Context, opts flag.Options) error {
-	if opts.GlobalOptions.ConfigFile == "bad-config" {
+	if opts.ConfigFile == "bad-config" {
```

Comment:
Is there any practical benefit to omitting the embedded field? Or is the rationale simply that because it can be omitted, we do so? I sometimes find it necessary to reference the embedded field explicitly for readability. So, I don’t think this is something that a linter should enforce. 

In particular, IDE “jump to definition” often lands you straight in the embedded struct’s field, which can obscure the relationship.  Of course, it might be helpful to define project conventions for when to omit the qualifier versus when to be explicit, but I don’t believe it’s worth enforcing one approach over the other at this time.

Reply:
>can also avoid NPE.

Can you elaborate on how it helps to avoid NPE? AFAIK, `u.Name` and `u.Profile.Name` are the same.
https://go.dev/play/p/5R3oJPPpqVD

Reply:
> In particular, IDE “jump to definition” often lands you straight in the embedded struct’s field, which can obscure the relationship.  Of course, it might be helpful to define project conventions for when to omit the qualifier versus when to be explicit, but I don’t believe it’s worth enforcing one approach over the other at this time.

I also agree with this as readability here is more important than brevity.